include/utils/kahan.hpp

```diff
@@ -43,4 +45,94 @@ T kahan(InputIt first, InputIt last, T init) {
     return sum;
 }
 
+template<class InputIt, class T>
+T kahanBabushka(InputIt first, InputIt last, T init) {
```

Comment:
I think we should either name this `kahan_babushka` (snake case for life), or we can remove the other `kahan` above, and call this one `kahan`, with a comment stating this is actually `Kahan-Babushka`. I am not sure there is a reason to have both; this one should be more accurate in all cases, and even though it is slightly more complicated, I don't think it will have a noticeable impact on performance.